frontend/apps/filemanager/filemanager.lua

```diff
@@ -248,6 +261,28 @@ function FileManager:init()
     self:handleEvent(Event:new("SetDimensions", self.dimen))
 end
 
+function FileManager:fileInformation(file)
+    --local book_stats = {}
+    local file_mode = lfs.attributes(file, "mode")
+    if file_mode == "file" then
+        local book_stats = DocSettings:open(file):readSetting('stats')
+        if book_stats ~= nil then
+            return FileManagerHistory:printFileInformation(book_stats)
+        else
+            local document = DocumentRegistry:openDocument(file)
+            ReaderUI:new{
+                dimen = Screen:getSize(),
+                document = document,
+            }
```

Comment:
Building a full readerui is a very expensive operation so it's better to avoid this if all you want is fetching the meta data. I assume you need this for page count? If so, could you try replace the ReaderUI creation code with the following?

```lua
if self.document.loadDocument then
    self.document:loadDocument()
    self.document:render()
end
```